tests/unit/web_tests.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import unittest
+
+from baseplate import web
+
+
+class IsSameDomainTests(unittest.TestCase):
+    def test_string_pattern(self):
+        self.assertTrue(web.is_same_domain("foo.com", "foo.com"))
+        self.assertFalse(web.is_same_domain("baz.com.com", "baz.com"))
+
+    def test_iterable_patterns(self):
+        patterns = {"foo.com", "bar.com"}
+        self.assertTrue(web.is_same_domain("foo.com", patterns))
+        self.assertTrue(web.is_same_domain("bar.com", patterns))
+        self.assertFalse(web.is_same_domain("baz.com", patterns))
+
+    def test_subdomain_matching(self):
+        self.assertTrue(web.is_same_domain("foo.com", ".foo.com"))
+        self.assertTrue(web.is_same_domain("bar.foo.com", ".foo.com"))
+        self.assertFalse(web.is_same_domain("bar.foo.com", "foo.com"))
```

Comment:
Can you add a test for the case where `patterns="bar.foo.com"` and `patterns=".bar.foo.com"`?